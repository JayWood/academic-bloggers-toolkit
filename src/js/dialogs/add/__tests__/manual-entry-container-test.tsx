/// <reference path="../../../../../lib/types/CSL.d.ts" />

import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { observable } from 'mobx';
import * as React from 'react';
import ManualEntryContainer from '../manual-entry-container';

const people = observable<CSL.TypedPerson>([]);
const manualData = observable.map<string>();
const errorMessage = observable('');
const mocks = {
    autoCite: jest.fn(),
    typeChange: jest.fn(),
};

const setup = ({ loading = false, itemType = 'webpage' } = {}) => {
    manualData.replace(new Map([['type', itemType]]));
    const component = shallow(
        <ManualEntryContainer
            loading={loading}
            manualData={manualData}
            errorMessage={errorMessage}
            people={people}
            {...mocks}
        />,
    );
    return {
        component,
    };
};

describe('<ManualEntryContainer />', () => {
    beforeEach(() => {
        people.replace([{ family: '', given: '', type: 'author' }]);
        errorMessage.set('');
        jest.resetAllMocks();
    });
    it('should match snapshots', () => {
        let { component } = setup();
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup({ loading: true }));
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup({ itemType: 'book' }));
        expect(toJSON(component)).toMatchSnapshot();

        ({ component } = setup({ itemType: 'article' }));
        expect(toJSON(component)).toMatchSnapshot();
    });
    it('should handle type change', () => {
        const { component } = setup();
        const select = component.find('select');
        select.simulate('change', { currentTarget: { value: 'book' } });
        expect(mocks.typeChange).toHaveBeenCalledTimes(1);
    });
    describe('wheel event tests', () => {
        const { component } = setup();
        const wheelDiv = component.find('.bounded-rect');
        let atTop: any = {};
        let inMiddle: any = {};
        let atBottom: any = {};
        beforeEach(() => {
            atTop = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 0,
                },
            };
            inMiddle = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 25,
                },
            };
            atBottom = {
                cancelable: true,
                deltaY: 0,
                currentTarget: {
                    clientHeight: 50,
                    scrollHeight: 100,
                    scrollTop: 50,
                },
            };
        });
        it('should cancel at top scrolling up', () => {
            atTop.deltaY = -5;
            wheelDiv.simulate('wheel', atTop);
            expect(atTop.cancelable).toBe(true);
        });
        it('should cancel at bottom scrolling down', () => {
            atBottom.deltaY = 5;
            wheelDiv.simulate('wheel', atBottom);
            expect(atBottom.cancelable).toBe(true);
        });
        it('should NOT cancel in middle scrolling up', () => {
            inMiddle.deltaY = -5;
            wheelDiv.simulate('wheel', inMiddle);
            expect(inMiddle.cancelable).toBe(false);
        });
        it('should NOT cancel in middle scrolling down', () => {
            inMiddle.deltaY = 5;
            wheelDiv.simulate('wheel', inMiddle);
            expect(inMiddle.cancelable).toBe(false);
        });
    });
});
