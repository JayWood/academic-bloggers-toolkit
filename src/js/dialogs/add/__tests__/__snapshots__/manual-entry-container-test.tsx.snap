// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ManualEntryContainer /> should match snapshots 1`] = `
<div>
  <div
    id="type-select-row"
  >
    <label
      htmlFor="type-select"
    >
      Citation Type
    </label>
    <select
      id="type-select"
      onChange={[Function]}
      value="webpage"
    >
      <option
        aria-selected={false}
        value="bill"
      >
        Bill
      </option>
      <option
        aria-selected={false}
        value="book"
      >
        Book
      </option>
      <option
        aria-selected={false}
        value="chapter"
      >
        Book Section
      </option>
      <option
        aria-selected={false}
        value="legal_case"
      >
        Case
      </option>
      <option
        aria-selected={false}
        value="paper-conference"
      >
        Conference Proceeding
      </option>
      <option
        aria-selected={false}
        value="entry-encyclopedia"
      >
        Encyclopedia Entry
      </option>
      <option
        aria-selected={false}
        value="motion_picture"
      >
        Film
      </option>
      <option
        aria-selected={false}
        value="speech"
      >
        Hearing
      </option>
      <option
        aria-selected={false}
        value="article-journal"
      >
        Journal Article
      </option>
      <option
        aria-selected={false}
        value="article-magazine"
      >
        Magazine Article
      </option>
      <option
        aria-selected={false}
        value="article-newspaper"
      >
        Newspaper Article
      </option>
      <option
        aria-selected={false}
        value="patent"
      >
        Patent
      </option>
      <option
        aria-selected={false}
        value="testing"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="testing2"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="report"
      >
        Report
      </option>
      <option
        aria-selected={false}
        value="legislation"
      >
        Statute
      </option>
      <option
        aria-selected={false}
        value="thesis"
      >
        Thesis
      </option>
      <option
        aria-selected={false}
        value="broadcast"
      >
        Television Broadcast
      </option>
      <option
        aria-selected={true}
        value="webpage"
      >
        Web Page
      </option>
    </select>
  </div>
  <AutoCite
    getter={[Function]}
    inputType="url"
    kind="webpage"
    placeholder="URL"
  />
  <div
    className="bounded-rect autocite"
    onWheel={[Function]}
  >
    <Callout
      dismiss={[Function]}
      intent="danger"
      isVisible={true}
    />
    <People
      citationType="webpage"
      people={
        Array [
          Object {
            "family": "",
            "given": "",
            "type": "author",
          },
        ]
      }
    />
    <MetaFields
      meta={
        Object {
          "type": "webpage",
        }
      }
    />
  </div>
  <style
    jsx={true}
  >
    
                        label {
                            margin-right: 10px;
                        }
                        select {
                            flex: auto;
                        }
                        #type-select-row {
                            display: flex;
                            padding: 0 10px 10px;
                            align-items: center;
                        }
                        .bounded-rect {
                            max-height: calc(100vh - 200px);
                            overflow-y: auto;
                            overflow-x: hidden;
                        }
                        .autocite {
                            max-height: calc(100vh - 250px);
                        }
                    
  </style>
</div>
`;

exports[`<ManualEntryContainer /> should match snapshots 2`] = `
<div>
  <Spinner
    bgColor="transparent"
    overlay={true}
    size="40px"
  />
  <div
    id="type-select-row"
  >
    <label
      htmlFor="type-select"
    >
      Citation Type
    </label>
    <select
      id="type-select"
      onChange={[Function]}
      value="webpage"
    >
      <option
        aria-selected={false}
        value="bill"
      >
        Bill
      </option>
      <option
        aria-selected={false}
        value="book"
      >
        Book
      </option>
      <option
        aria-selected={false}
        value="chapter"
      >
        Book Section
      </option>
      <option
        aria-selected={false}
        value="legal_case"
      >
        Case
      </option>
      <option
        aria-selected={false}
        value="paper-conference"
      >
        Conference Proceeding
      </option>
      <option
        aria-selected={false}
        value="entry-encyclopedia"
      >
        Encyclopedia Entry
      </option>
      <option
        aria-selected={false}
        value="motion_picture"
      >
        Film
      </option>
      <option
        aria-selected={false}
        value="speech"
      >
        Hearing
      </option>
      <option
        aria-selected={false}
        value="article-journal"
      >
        Journal Article
      </option>
      <option
        aria-selected={false}
        value="article-magazine"
      >
        Magazine Article
      </option>
      <option
        aria-selected={false}
        value="article-newspaper"
      >
        Newspaper Article
      </option>
      <option
        aria-selected={false}
        value="patent"
      >
        Patent
      </option>
      <option
        aria-selected={false}
        value="testing"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="testing2"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="report"
      >
        Report
      </option>
      <option
        aria-selected={false}
        value="legislation"
      >
        Statute
      </option>
      <option
        aria-selected={false}
        value="thesis"
      >
        Thesis
      </option>
      <option
        aria-selected={false}
        value="broadcast"
      >
        Television Broadcast
      </option>
      <option
        aria-selected={true}
        value="webpage"
      >
        Web Page
      </option>
    </select>
  </div>
  <AutoCite
    getter={[Function]}
    inputType="url"
    kind="webpage"
    placeholder="URL"
  />
  <div
    className="bounded-rect autocite"
    onWheel={[Function]}
  >
    <Callout
      dismiss={[Function]}
      intent="danger"
      isVisible={true}
    />
    <People
      citationType="webpage"
      people={
        Array [
          Object {
            "family": "",
            "given": "",
            "type": "author",
          },
        ]
      }
    />
    <MetaFields
      meta={
        Object {
          "type": "webpage",
        }
      }
    />
  </div>
  <style
    jsx={true}
  >
    
                        label {
                            margin-right: 10px;
                        }
                        select {
                            flex: auto;
                        }
                        #type-select-row {
                            display: flex;
                            padding: 0 10px 10px;
                            align-items: center;
                        }
                        .bounded-rect {
                            max-height: calc(100vh - 200px);
                            overflow-y: auto;
                            overflow-x: hidden;
                        }
                        .autocite {
                            max-height: calc(100vh - 250px);
                        }
                    
  </style>
</div>
`;

exports[`<ManualEntryContainer /> should match snapshots 3`] = `
<div>
  <div
    id="type-select-row"
  >
    <label
      htmlFor="type-select"
    >
      Citation Type
    </label>
    <select
      id="type-select"
      onChange={[Function]}
      value="book"
    >
      <option
        aria-selected={false}
        value="bill"
      >
        Bill
      </option>
      <option
        aria-selected={true}
        value="book"
      >
        Book
      </option>
      <option
        aria-selected={false}
        value="chapter"
      >
        Book Section
      </option>
      <option
        aria-selected={false}
        value="legal_case"
      >
        Case
      </option>
      <option
        aria-selected={false}
        value="paper-conference"
      >
        Conference Proceeding
      </option>
      <option
        aria-selected={false}
        value="entry-encyclopedia"
      >
        Encyclopedia Entry
      </option>
      <option
        aria-selected={false}
        value="motion_picture"
      >
        Film
      </option>
      <option
        aria-selected={false}
        value="speech"
      >
        Hearing
      </option>
      <option
        aria-selected={false}
        value="article-journal"
      >
        Journal Article
      </option>
      <option
        aria-selected={false}
        value="article-magazine"
      >
        Magazine Article
      </option>
      <option
        aria-selected={false}
        value="article-newspaper"
      >
        Newspaper Article
      </option>
      <option
        aria-selected={false}
        value="patent"
      >
        Patent
      </option>
      <option
        aria-selected={false}
        value="testing"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="testing2"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="report"
      >
        Report
      </option>
      <option
        aria-selected={false}
        value="legislation"
      >
        Statute
      </option>
      <option
        aria-selected={false}
        value="thesis"
      >
        Thesis
      </option>
      <option
        aria-selected={false}
        value="broadcast"
      >
        Television Broadcast
      </option>
      <option
        aria-selected={false}
        value="webpage"
      >
        Web Page
      </option>
    </select>
  </div>
  <AutoCite
    getter={[Function]}
    inputType="text"
    kind="book"
    pattern="(?:[dxX]-?){10}|(?:[dxX]-?){13}"
    placeholder="ISBN"
  />
  <div
    className="bounded-rect autocite"
    onWheel={[Function]}
  >
    <Callout
      dismiss={[Function]}
      intent="danger"
      isVisible={true}
    />
    <People
      citationType="book"
      people={
        Array [
          Object {
            "family": "",
            "given": "",
            "type": "author",
          },
        ]
      }
    />
    <MetaFields
      meta={
        Object {
          "type": "book",
        }
      }
    />
  </div>
  <style
    jsx={true}
  >
    
                        label {
                            margin-right: 10px;
                        }
                        select {
                            flex: auto;
                        }
                        #type-select-row {
                            display: flex;
                            padding: 0 10px 10px;
                            align-items: center;
                        }
                        .bounded-rect {
                            max-height: calc(100vh - 200px);
                            overflow-y: auto;
                            overflow-x: hidden;
                        }
                        .autocite {
                            max-height: calc(100vh - 250px);
                        }
                    
  </style>
</div>
`;

exports[`<ManualEntryContainer /> should match snapshots 4`] = `
<div>
  <div
    id="type-select-row"
  >
    <label
      htmlFor="type-select"
    >
      Citation Type
    </label>
    <select
      id="type-select"
      onChange={[Function]}
      value="article"
    >
      <option
        aria-selected={false}
        value="bill"
      >
        Bill
      </option>
      <option
        aria-selected={false}
        value="book"
      >
        Book
      </option>
      <option
        aria-selected={false}
        value="chapter"
      >
        Book Section
      </option>
      <option
        aria-selected={false}
        value="legal_case"
      >
        Case
      </option>
      <option
        aria-selected={false}
        value="paper-conference"
      >
        Conference Proceeding
      </option>
      <option
        aria-selected={false}
        value="entry-encyclopedia"
      >
        Encyclopedia Entry
      </option>
      <option
        aria-selected={false}
        value="motion_picture"
      >
        Film
      </option>
      <option
        aria-selected={false}
        value="speech"
      >
        Hearing
      </option>
      <option
        aria-selected={false}
        value="article-journal"
      >
        Journal Article
      </option>
      <option
        aria-selected={false}
        value="article-magazine"
      >
        Magazine Article
      </option>
      <option
        aria-selected={false}
        value="article-newspaper"
      >
        Newspaper Article
      </option>
      <option
        aria-selected={false}
        value="patent"
      >
        Patent
      </option>
      <option
        aria-selected={false}
        value="testing"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="testing2"
      >
        Test not in use
      </option>
      <option
        aria-selected={false}
        value="report"
      >
        Report
      </option>
      <option
        aria-selected={false}
        value="legislation"
      >
        Statute
      </option>
      <option
        aria-selected={false}
        value="thesis"
      >
        Thesis
      </option>
      <option
        aria-selected={false}
        value="broadcast"
      >
        Television Broadcast
      </option>
      <option
        aria-selected={false}
        value="webpage"
      >
        Web Page
      </option>
    </select>
  </div>
  <div
    className="bounded-rect"
    onWheel={[Function]}
  >
    <Callout
      dismiss={[Function]}
      intent="danger"
      isVisible={true}
    />
    <MetaFields
      meta={
        Object {
          "type": "article",
        }
      }
    />
  </div>
  <style
    jsx={true}
  >
    
                        label {
                            margin-right: 10px;
                        }
                        select {
                            flex: auto;
                        }
                        #type-select-row {
                            display: flex;
                            padding: 0 10px 10px;
                            align-items: center;
                        }
                        .bounded-rect {
                            max-height: calc(100vh - 200px);
                            overflow-y: auto;
                            overflow-x: hidden;
                        }
                        .autocite {
                            max-height: calc(100vh - 250px);
                        }
                    
  </style>
</div>
`;
